v1/runGame.php

```diff
@@ -11,83 +11,39 @@
 include_once("lib/external/vendor/autoload.php");
 
 foreach (glob('lib/output/*.php') as $file) include( $file );
+foreach (glob('lib/input/*.php') as $file) include( $file );
 
 use Ulrichsg\Getopt;
 $options = new Getopt(array(
-            array('o','output',Getopt::REQUIRED_ARGUMENT,'Output in console, as png or gif'),
-            array('w','width',Getopt::REQUIRED_ARGUMENT,'Length of x-axis of the gamefield 1-*'),
-            array('h','height',Getopt::REQUIRED_ARGUMENT,'Length of the y-axis of the gamefield 1-*'),
-            array('c','numCycles',Getopt::REQUIRED_ARGUMENT, 'Number of rounds the game should play 1-*'),
-            array('t','type',Getopt::REQUIRED_ARGUMENT, 'Figure that should be set, blinker, glider, or lws'),
-            array('H', 'help', Getopt::NO_ARGUMENT)
+    array('i','input',Getopt::REQUIRED_ARGUMENT,'Input type (for now only txt is implemented'),
+    array('f','filepath',Getopt::OPTIONAL_ARGUMENT,'Path to txt file(only required if txt is choosen as input'),
+    array('o','output',Getopt::REQUIRED_ARGUMENT,'Output in console, as png or gif'),
+    array('c','numCycles',Getopt::REQUIRED_ARGUMENT, 'Number of rounds the game should play 1-*'),
+    array('H', 'help', Getopt::NO_ARGUMENT)
 
 ));
 
 $options->parse();
 
-if ($options->getOption('width')) $x = $options->getOption('h');
-else echo "Argument --width is missing\n";
-
-if ($options->getOption('height')) $y = $options->getOption('h');
-else echo "Argument --height is missing\n";
-
-if (isset($x) && isset($y))
+if ($options->getOption('input'))
 {
-    $gameField = new GameField($x,$y);
-
-    if($options->getOption('type')) {
-
-        switch ($options->getOption('type'))
-        {
-            case "blinker":
-                $gameField->getCellByCoords(3, 2)->life();
-                $gameField->getCellByCoords(3, 3)->life();
-                $gameField->getCellByCoords(3, 4)->life();
-                break;
-
-            case "glider":
-                $gameField->getCellByCoords(2, 1)->life();
-                $gameField->getCellByCoords(3, 2)->life();
-                $gameField->getCellByCoords(3, 3)->life();
-                $gameField->getCellByCoords(2, 3)->life();
-                $gameField->getCellByCoords(1, 3)->life();
-                break;
+    if ($options->getOption("filepath")) $config = array('filePath' => $options->getOption('filepath'));
+    $inputClass = $options->getOption("input")."Input";
+    $input = new $inputClass($config);
```

Comment:
Yes but i wrote as default value of the parameter NULL in the baseclass, so NULL is delivered. This is because (i said it already above) for example a console output plugin wouldn't need the config.